pkg/config/reload.go

```diff
@@ -148,6 +150,16 @@ func (c *Config) ReloadPauseImage(newConfig *Config) error {
 // ReloadPinnedImages updates the PinnedImages with the provided `newConfig`.
 // The method print log in case of any updates.
 func (c *Config) ReloadPinnedImages(newConfig *Config) {
+	if len(newConfig.PinnedImages) == 0 {
+		c.PinnedImages = newConfig.PinnedImages
+		return
+	}
```

Comment:
Setting the value to an empty slice might be more explicit.

```suggestion
	if len(newConfig.PinnedImages) == 0 {
		c.PinnedImages = []string{}
		return
	}
```